test/utils.spec.js

```diff
@@ -34,33 +34,33 @@ describe('utils', function() {
     });
 
     it('Handles deeply nested object', function() {
-      expect(decodeURI(utils.stringifyRequestData({
+      expect(utils.stringifyRequestData({
         a: {
           b: {
             c: {
               d: 2,
             },
           },
         },
-      }))).to.equal('a[b][c][d]=2');
+      })).to.equal('a[b][c][d]=2');
     });
 
     it('Handles arrays of objects', function() {
-      expect(decodeURI(utils.stringifyRequestData({
+      expect(utils.stringifyRequestData({
         a: [
           {b: 'c'},
           {b: 'd'},
         ],
-      }))).to.equal('a[][b]=c&a[][b]=d');
+      })).to.equal('a[0][b]=c&a[1][b]=d');
     })
 
     it('Handles indexed arrays', function() {
-      expect(decodeURI(utils.stringifyRequestData({
+      expect(utils.stringifyRequestData({
         a: {
-          0: 'c',
-          1: 'd',
+          0: {b: 'c'},
+          1: {b: 'd'},
         },
-      }))).to.equal('a[0]=c&a[1]=d');
+      })).to.equal('a[0][b]=c&a[1][b]=d');
```

Comment:
curious, why this change?

Reply:
Mainly to illustrate that a list of hashes and an integer-indexed hash of hashes have the same encoded representation.